lib/pushmi_pullyu/aip_creator.rb

```diff
@@ -0,0 +1,73 @@
+require 'pushmi_pullyu'
+require 'pushmi_pullyu/aip_downloader'
+require 'bagit'
+require 'archive/tar/minitar'
+
+class PushmiPullyu::AipCreator
+
+  attr_reader :noid, :config, :logger,
+              :workdir, :basedir, :tar_filename
+
+  def initialize(noid, config = nil, logger = nil)
+    @noid = noid
+
+    @config = config || PushmiPullyu.options
+    @logger = logger || PushmiPullyu.logger
```

Comment:
I guess part of it is to make testing easier. Since we don't have production/development/test environment switching, this seemed like a good way to go to test individual classes/objects in isolation. See for example, overriding the config here: https://github.com/ualbertalib/pushmi_pullyu/blob/cwant/issue_2_pull_fedora_solr3/spec/pushmi_pullyu/fedora_object_fetcher_spec.rb#L4